src/charm.py

```diff
@@ -61,6 +65,19 @@ def __init__(self, *args: typing.Any):
         self.framework.observe(self.on.jenkins_pebble_ready, self._on_jenkins_pebble_ready)
         self.framework.observe(self.on.update_status, self._on_update_status)
 
+    @property
+    def is_storage_ready(self) -> bool:
+        """Return whether the Jenkins home storage is mounted.
+
+        Returns:
+            True if storage is mounted, False otherwise.
+        """
+        container = self.unit.get_container(self.state.jenkins_service_name)
+        if not container.can_connect():
+            return False
+        mount_info: str = container.pull("/proc/mounts").read()
+        return str(jenkins.HOME_PATH) in mount_info
```

Comment:
Can't we make this a shared function that takes the container as an argument so we don't need to pass the charm around to other classes?

Reply:
@jdkandersson  Out of curiosity, would there be a reason why passing charm around would not be preferred? I was thinking that `charm.py` could provide utility functions that could be shared around so that such functions don't need to be repeated and be readily available around the observers.

Reply:
Because of circular dependencies which is why the charm is no longer typed as a base charm and instead as the specific charm, so it depends on its creator which is difficult to reason about

Reply:
The circular dependency is due to the typing - no runtime dependency - I'm not quite sure I understand why coupling here would be difficult to reason about.

Reply:
Updated, thank you!